### DISCUSSION TITLE
Missing functionality

### DISCUSSION THREAD

**Discussion by adevade** [alpinejs/alpine#2782] [7. Bugs (archived)] [answered]
I don't seem to get the `x-on:click="return confirm('...')"` to work anymore. It was implemented some time ago, in a previous version.

Issue: https://github.com/alpinejs/alpine/issues/150
PR: https://github.com/alpinejs/alpine/pull/162

The native `onclick="return confirm('...')"` is working as expected.

**Comment by leobm** [comment]
I think your problem is the return keyword.
    
this works here:
```
<div x-data="{ 'result': false}">

            <div x-text="result"></div>
            <p x-on:click="result = confirm('hello')">BLAHHH</p>
    </div>
```

**Reply by adevade** [reply]
Hmm, I'm using it on a `<a>` tag so maybe that's the problem?

The link should open a confirm dialog, and only continue to the link if "OK" is pressed.
It works well with the native `onlick="return confirm('Do you want to open the link?')"`, but not with any of the other alternatives.

I created a isolated example here: https://codepen.io/adevade/pen/abEpmom

**Reply by leobm** [reply]
you can do e.g. the following if you want to prevent the link click via the confirm dialog 

```html
    <div x-data="{
        doConfirm(evt) {
            if (!confirm('hello')) {
            evt.preventDefault();
            }
        }
    }">

        <a href="http://www.google.de" x-on:click="doConfirm($event)">BLAHHH</a>
    </div>
```

**Reply by leobm** [reply]
I have just written a very simple directive that may abstract my previous example away
https://codepen.io/leobm/pen/dyJNOdB
```
<!DOCTYPE html>
<html lang="en">
<head>
    <script src="//unpkg.com/alpinejs" defer></script>
    <script>
        document.addEventListener('alpine:init', () => {

            Alpine.directive('confirm', (el, {value, modifiers, expression}, {effect, evaluate, cleanup}) => {
                let handler = (evt) => {
                    if (!window.confirm(expression)) {
                        evt.preventDefault();
                    }
                };
                el.addEventListener('click', handler);
                cleanup(() => {
                    window.removeEventListener('click', handler)
                });
            });
        });
    </script>
</head>
<body>
    <div x-data="{}">
        <a href="http://www.google.de" x-confirm="Do you want to open the link?">BLAHHH</a>
    </div>
</body>
</html>
```

**Reply by leobm** [reply]
maybe, it might be better to evaluate the expression to in the directive if the confirm text itself should come from the x-data model, for example.

**Comment by SimoTod** [comment, marked as answer]
That pr and issue you linked were for V2.
See https://alpinejs.dev/upgrade-guide#no-false-return-from-event-handlers

**Reply by adevade** [reply]
Thank you! 👍 I had missed that! I'll just keep using `onclick=""` then.
